### PR TITLE
Patch for using DB Rider with MS SQL Server

### PR DESCRIPTION
**Pull request by ArtemIurkov**
Added IDENTITY_INSERT SeedStrategy to avoid following error:
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Cannot insert explicit value for identity column in table 'xxxxx' when IDENTITY_INSERT is set to OFF.

### CODE REVIEW

**Review by rmpestano — CHANGES_REQUESTED** [review]
Really nice, thank you!

Does that mean that we can rewrite this integration test to use the new strategy or I got it wrong? 

If so can you update the test or add a new one?

Thank you very much!

**Comment by rmpestano** [comment]
Also I noticed that we have a build error, can you check?

Screenshot 2021-08-18 at 13 24 25

File: rider-core/src/main/java/com/github/database/rider/core/api/dataset/SeedStrategy.java [hunk 1]
```diff
@@ -28,7 +28,8 @@
     TRUNCATE_INSERT(new CompositeOperation(DatabaseOperation.TRUNCATE_TABLE, DatabaseOperation.INSERT)),
     INSERT(DatabaseOperation.INSERT),
     REFRESH(DatabaseOperation.REFRESH),
-    UPDATE(DatabaseOperation.UPDATE);
+    UPDATE(DatabaseOperation.UPDATE),
+    IDENTITY_INSERT(InsertIdentityOperation.CLEAN_INSERT);
```
Comment (rmpestano) [inline rider-core/src/main/java/com/github/database/rider/core/api/dataset/SeedStrategy.java:32]: Can you update the javadoc with the new strategy?

Comment (artem-iurkov-slidepresenter) [reply rider-core/src/main/java/com/github/database/rider/core/api/dataset/SeedStrategy.java:35]: Done

**Comment by ArtemIurkov** [comment]
> Also I noticed that we have a build error, can you check?
> 
> Screenshot 2021-08-18 at 13 24 25

Thank you, fixed

**Comment by ArtemIurkov** [comment]
> Really nice, thank you!
> 
> Does that mean that we can rewrite [this integration test](https://github.com/database-rider/database-rider/blob/master/rider-core/src/test/java/com/github/database/rider/core/MsSQLDatabaseIdentitySupportIt.java) to use the new strategy or I got it wrong?
> 
> If so can you update the test or add a new one?
> 
> Thank you very much!

Not sure my qualification is enough to do it.
Maybe can do something if you provide me some hints.

**Comment by rmpestano** [comment]
> Maybe can do something if you provide me some hints.

I think adding a [test here](https://github.com/database-rider/database-rider/blob/ea2068a0773c1fe574267f0b42bf01d6f9c14ddb/rider-core/src/test/java/com/github/database/rider/core/MsSQLDatabaseIt.java#L23) using the new seeding strategy would be enough, WDYT?

**Comment by ArtemIurkov** [comment]
> > Maybe can do something if you provide me some hints.
> 
> I think adding a [test here](https://github.com/database-rider/database-rider/blob/ea2068a0773c1fe574267f0b42bf01d6f9c14ddb/rider-core/src/test/java/com/github/database/rider/core/MsSQLDatabaseIt.java#L23) using the new seeding strategy would be enough, WDYT?

I added new test that uses new strategy

**Comment by rmpestano** [comment]
> I added new test that uses new strategy

Thank you! Let's wait the build to pass and then merge. After that we can release a new version with your contribution

File: rider-core/src/test/java/com/github/database/rider/core/MsSQLDatabaseIt.java [hunk 2]
```diff
@@ -64,7 +73,7 @@ public void shouldSeedDataSetDisablingConstraints() {
     @DBUnit(leakHunter = true, caseSensitiveTableNames = true, escapePattern = "\"?\"")
     public void shouldFindConnectionLeak() throws SQLException {
         exception.expect(LeakHunterException.class);
-        exception.expectMessage("Execution of method shouldFindConnectionLeak left 1 open connection(s).");
+        exception.expectMessage("Execution of method shouldFindConnectionLeak left 2 open connection(s).");
```
Comment (rmpestano) [inline rider-core/src/test/java/com/github/database/rider/core/MsSQLDatabaseIt.java:76]: We can keep as `1` open connection here, this test with leak hunter is failing eventually and need to have a deeper look why. So please just keep as it was before

**Review by rmpestano — CHANGES_REQUESTED** [review]

File: rider-core/src/test/java/com/github/database/rider/core/MsSQLDatabaseIt.java (hunk 2 above)
Comment (ArtemIurkov) [reply rider-core/src/test/java/com/github/database/rider/core/MsSQLDatabaseIt.java]: Done

**Review by rmpestano — APPROVED** [review]

**Comment by ArtemIurkov** [comment]
> > I added new test that uses new strategy
> 
> Thank you! Let's wait the build to pass and then merge. After that we can release a new version with your contribution

Thank you Rafael!
What version of DB Rider will include this changes?

**Comment by rmpestano** [comment]
> What version of DB Rider will include this changes?
> When you can release it?

It will be in `1.29.0` which should be already in maven central. 

Thank you for this contribution!

**Comment by artem-iurkov-slidepresenter** [comment]
> > What version of DB Rider will include this changes?
> > When you can release it?
> 
> It will be in `1.29.0` which should be already in maven central.
> 
> Thank you for this contribution!

Thank you for a quick release!!!

**Pull request merged**